enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingProcessRepository.java

```diff
@@ -40,72 +42,133 @@
 @Repository
 public interface OnboardingProcessRepository extends CrudRepository<OnboardingProcessEntity, String> {
 
+    /**
+     * Find an onboarding process using process ID. Lock this process using PESSIMISTIC_WRITE lock
+     * until the end of the transaction.
+     *
+     * @param processId Onboarding process identifier.
+     * @return Optional onboarding process.
+     */
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.id = :processId")
+    Optional<OnboardingProcessEntity> findByIdWithLock(String processId);
+
+    /**
+     * Find an existing process using identification data and status. Lock this process using PESSIMISTIC_WRITE lock
+     * until the end of the transaction.
+     *
+     * @param identificationData Identification data.
+     * @param status Process status.
+     * @return Optional onboarding process.
+     */
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
     @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.status = :status AND p.identificationData = :identificationData")
-    Optional<OnboardingProcessEntity> findExistingProcessByIdentificationData(String identificationData, OnboardingStatus status);
+    Optional<OnboardingProcessEntity> findByIdentificationDataAndStatusWithLock(String identificationData, OnboardingStatus status);
 
+    /**
+     * Find an existing process by activation identifier and process status. The process is not locked.
+     *
+     * @param activationId Activation identifier.
+     * @param status Onboarding process status.
+     * @return Optional onboarding process.
```

Comment:
Nice naming. If you want to be super perfect, you may cross reverence non/locking methods via `@see`.

Reply:
Ok, good idea.